quinn-interop/src/client.rs

```diff
@@ -168,21 +170,25 @@ async fn hq_download_all(
     conn: quinn::Connection,
     requests: &[http::Uri],
 ) -> Result<(), Box<dyn std::error::Error>> {
-    // TODO spawn
-    future::try_join_all(requests.into_iter().cloned().map(move |req| {
+    for req in requests.into_iter().cloned() {
```

Comment:
Nit: this commit has kind of a lot going on, maybe make this three commits?

- Extract `hq_download()`
- Extract `h3_download()`
- Get rid of futures dependency